drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/EditableTextBox.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.testscenario.client;
+
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.Widget;
+import org.uberfire.client.callbacks.Callback;
+
+public class EditableTextBox
+        implements IsWidget {
+
+    private final EditableTextBoxView view;
+    private Callback<String> changed;
+
+    public EditableTextBox(final Callback<String> changed,
+                           final EditableTextBoxView view,
+                           final String fieldName,
+                           final String initialValue) {
+        this.changed = changed;
+        this.view = view;
+        view.setPresenter(this);
+
+        changed.callback(view.getText());
```

Comment:
This should not be needed.

1. You add a `ValueChangeHandler` [here](https://github.com/kiegroup/drools-wb/pull/1231/files#diff-694563eaf1d207c6a6706ca0e0f9ab77R39).
2. When the value is set [here](https://github.com/kiegroup/drools-wb/pull/1231/files#diff-694563eaf1d207c6a6706ca0e0f9ab77R49), if it is invalid, the `ValueChangeHandler` attached above should be called.

So, provided, you call `view.setPresenter(this)` **before** `view.setText(initialValue)` (as you already do) then the existing mechanism to update the model with the default/valid value should execute.

Reply:
Right I actually meant to do this, but now this gives me another puzzle on how to test this in a way that makes sense. I'll figure something out.